test/python/test_instructions.py

```diff
@@ -585,6 +586,106 @@ def test_sign_level_authorized(
         with StatusCode.WRONG_VALUES.expected():
             client.sign_message(account, message_2)
 
+
+@pytest.mark.parametrize("account", ACCOUNTS)
+@pytest.mark.parametrize("with_hash", [False, True])
+def test_sign_delegation(
+        account: Account,
+        with_hash: bool,
+        tezos_navigator: TezosNavigator) -> None:
+    """Test the SIGN(_WITH_HASH) instruction on delegation."""
+    snap_path = Path(f"{account}")
+
+    tezos_navigator.setup_app_context(
+        account,
+        DEFAULT_CHAIN_ID,
+        main_hwm=Hwm(0),
+        test_hwm=Hwm(0)
+    )
+
+    delegation = Delegation(
+        delegate=account.public_key_hash,
+        source=account.public_key_hash,
+    )
+
+    raw_delegation = delegation.forge()
+
+    if with_hash:
```

Comment:
Shouldn't it be oppsite? if with_hash is true, we should sign with hash.

Reply:
Fixed here: https://github.com/trilitech/ledger-app-tezos-baking/pull/24/commits/25cdc8d279a03be5050f5da1e213410ab36001d3